### PR TITLE
Support hiding battles from /cmd userdetails

### PR DESCRIPTION
**Pull request by AnnikaCodes**
I've heard that some global staff want this feature as a way to prevent people from joining battles by clicking their usercard, without full-on hiding all battles they play in.

### CODE REVIEW

**Comment by Zarel** [comment]
:/ We have to be pretty careful with the name for this function so it doesn't get confused with other similar functionality.

**Comment by AnnikaCodes** [comment]
> :/ We have to be pretty careful with the name for this function so it doesn't get confused with other similar functionality.

Yeah, /nodisplaybattles was the best one I could think of, but I'm open to suggestions.

**Comment by Zarel** [comment]
Generally I default to "really long name" if it's something that can easily be misunderstood.

Something like `/hidebattlesfromtrainercard`

**Comment by Zarel** [comment]
Probably alias `/hidebattlesfromusercard`

**Comment by Zarel** [comment]
I would also call the variable `settings.hideBattlesFromTrainerCard`.

**Comment by Zarel** [comment]
I like this! @scheibo do you have opinions? Otherwise I'm merging.

File: server/chat-commands/core.ts [hunk 1]
```diff
@@ -1321,6 +1321,19 @@ export const commands: ChatCommands = {
 		});
 	},
 
+	showbattlesinusercard: 'hidebattlesfromtrainercard',
+	hidebattlesfromusercard: 'hidebattlesfromtrainercard',
+	showbattlesintrainercard: 'hidebattlesfromtrainercard',
+	hidebattlesfromtrainercard(target, room, user, connection, cmd) {
+		const shouldHide = cmd.includes('hide');
+		user.settings.hideBattlesFromTrainerCard = shouldHide;
```
Comment (scheibo) [inline server/chat-commands/core.ts:1329]: You need to call user.update() or whatever to propagate the setting via |updateuser| to the client

**Pull request merged**